OrbitGl/TracepointTrack.cpp

```diff
@@ -53,13 +69,35 @@ void TracepointTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick, Pic
 
   const Color kWhite(255, 255, 255, 255);
 
-  for (auto it = tracepoints.lower_bound(min_tick); it != tracepoints.upper_bound(max_tick); ++it) {
-    uint64_t time = it->first;
-    if (time < max_tick) {
-      Vec2 pos(time_graph_->GetWorldFromTick(time), pos_[1]);
-      batcher->AddVerticalLine(pos, -track_height, z, kWhite);
-    } else {
-      return;
+  const Color kGreenSelection(0, 255, 0, 255);
+
+  if (!picking) {
+    for (auto it = tracepoints.lower_bound(min_tick); it != tracepoints.upper_bound(max_tick);
+         ++it) {
+      uint64_t time = it->first;
+      if (time < max_tick) {
```

Comment:
You can remove the `if` and the `else` branch with the early return. The loop is running from lower_bound to upper_bound so you should only get `time`s between `min_tick `and `max_tick`.

Reply:
Yes, you are right. Thank you for noticing!